dask/highlevelgraph.py

```diff
@@ -870,6 +870,49 @@ def merge(cls, *graphs):
         return cls(layers, dependencies)
 
     def visualize(self, filename="dask-hlg.svg", format=None, **kwargs):
+        """
+        Visualize a dask high level graph.
+
+        Requires ``graphviz`` to be installed. All options that are not the dask
+        graph(s) should be passed as keyword arguments.
+
+        Parameters
+        ----------
+        filename : str or None, optional
+            The name of the file to write to disk. If the provided `filename`
+            doesn't include an extension, '.svg' will be used by default.
```

Comment:
Doesn't the extension you get depend on the `format` argument?

Reply:
It also depends on `filename`.

https://github.com/dask/dask/blob/88a13f77f374a11425961c074f4e579f3befe721/dask/dot.py#L282-L283

but, `format` would overwrite the format in the `filename`.
`x.dask.visualize(filename="thishouldbesvg.svg", format="png")`
would `thishouldbesvg.svg.png`!

Reply:
I don't know what the code produces (you can test), but the wording here implies that when you give a filename without any extension at all, ".svg" is appended - which I suspect is not necessarily true.

Reply:
Oh, you are indeed right!

Reply:
```py
import dask.array as da

x = da.ones((10, 10), chunks=(5, 5))
x = x + 100
x = x * 100

x.dask.visualize(filename="isthissvg")
```
produces `isthissvg.png`

Reply:
Wow, I found some tiny flaws in the filename/format system.
I will create an issue on it once I confirm them.

Reply:
For now, I have just changed 
`'.svg' will be used by default.`
to 
`'.png' will be used by default.`
which is true.

_I will make this doc even better in another PR which fixes the filename/format situation for HLGs._

Is this fine for now?

Reply:
>  '.png' will be used by default.

Is it true regardless of the `format` argument?

> I will make this doc even better in another PR

yes, that's fine.

Reply:
> Is it true regardless of the format argument?
yes!

Reply:
EDIT: format would overwrite the filename (if the filename has an extension attached or not)

```py
x.dask.visualize(filename="whatextension.png", format='pdf') # whatextension.png.pdf
x.dask.visualize(filename="whatextension", format=None) # whatextension.png
```